discovery/nomad/nomad.go

```diff
@@ -171,46 +181,82 @@ func (d *Discovery) refresh(context.Context) ([]*targetgroup.Group, error) {
 	opts := &nomad.QueryOptions{
 		AllowStale: d.allowStale,
 	}
-	stubs, _, err := d.client.Services().List(opts)
+	tg := &targetgroup.Group{
+		Source: "Nomad",
+	}
+	serviceNames, err := d.getServiceNames()
 	if err != nil {
 		d.metrics.failuresCount.Inc()
-		return nil, err
+		return nil, fmt.Errorf("failed to get service names from nomad: %w", err)
 	}
+	if len(serviceNames) == 0 {
+		level.Warn(d.logger).Log("msg", "nomad service discovery did not return any services to be scrapped")
+		return []*targetgroup.Group{}, nil
+	}
+	level.Debug(d.logger).Log("msg", "services found through nomad service discovery", "service_names", strings.Join(serviceNames, ","))
```

Comment:
Do we need this?

Reply:
When I diagnose why service discovery isn't working in prometheus, it can frequently be poking at a black box. Logs like this help an operator determine whether a service was found or not. But I don't have strong feelings about it so i'll take it out.